zingo-sync/src/scanner.rs

```diff
@@ -0,0 +1,121 @@
+use std::collections::HashSet;
+
+use orchard::keys::Scope;
+use tokio::sync::mpsc::UnboundedSender;
+use zcash_client_backend::{
+    data_api::scanning::ScanRange, proto::compact_formats::CompactBlock, scanning::ScanningKeys,
+    PoolType, ShieldedProtocol,
+};
+use zcash_primitives::{consensus::Parameters, transaction::TxId, zip32::AccountId};
+
+use crate::{
+    client::{get_compact_block_range, FetchRequest},
+    primitives::OutputId,
+};
+
+use self::runners::BatchRunners;
+
+pub(crate) mod runners;
+
+pub(crate) async fn scanner<P>(
```

Comment:
this is confusing because scanner is a noun. is there a verb way to describe what this does?

Reply:
yeah i agree, i have changed this in following PR to scan :)